core/modules/wmgr/test/wmgrDbTest.py

```diff
@@ -0,0 +1,228 @@
+
+    def test20ChunkCreateDelete(self):
+        """ Test for database creation/listing/deletion. """
+
+        # make semi-random database name
+        dbName = self.dbName
+        tblName = "ObjectX"
+        tablesURI = '/dbs/' + dbName + '/tables'
+        chunksURI = tablesURI + '/' + tblName + '/chunks'
+
+        # create database/table first
+        response = self._getJson(self.app.post('/dbs', data=dict(db=dbName)), 201)
+        options = dict(table=tblName, schemaSource="request",
+                       schema="CREATE TABLE " + tblName + " (X INT, Y INT)")
+        response = self._getJson(self.app.post(tablesURI, data=options), 201)
+
+        # create few chunks
+        copt = dict(chunkId=10)
+        response = self._getJson(self.app.post(chunksURI, data=copt), 201)
+        self.assertEqual(response['result']['chunkId'], 10)
+        self.assertEqual(response['result']['uri'], chunksURI + '/10')
+        self.assertEqual(response['result']['chunkTable'], True)
+        self.assertEqual(response['result']['overlapTable'], True)
+        copt = dict(chunkId=20, overlapFlag='False')
+        response = self._getJson(self.app.post(chunksURI, data=copt), 201)
+        self.assertEqual(response['result']['chunkId'], 20)
+        self.assertEqual(response['result']['uri'], chunksURI + '/20')
+        self.assertEqual(response['result']['chunkTable'], True)
+        self.assertEqual(response['result']['overlapTable'], False)
+        copt = dict(chunkId=30, overlapFlag=True)
+        response = self._getJson(self.app.post(chunksURI, data=copt), 201)
+        self.assertEqual(response['result']['chunkId'], 30)
+        self.assertEqual(response['result']['uri'], chunksURI + '/30')
+        self.assertEqual(response['result']['chunkTable'], True)
+        self.assertEqual(response['result']['overlapTable'], True)
+
+        # drop table, without options this should delete main table but leave all chunks
+        response = self._getJson(self.app.delete(tablesURI + '/' + tblName))
+
+        # Get table list
+        response = self._getJson(self.app.get(tablesURI))
+        tables = set(obj['name'] for obj in response['results'])
+        expected = set(['ObjectX_10', 'ObjectX_20', 'ObjectX_30', 'ObjectXFullOverlap_10', 'ObjectXFullOverlap_30'])
+        self.assertEqual(tables, expected)
+
+        # re-create table
+        response = self._getJson(self.app.post(tablesURI, data=options), 201)
+
+        # now drop with dropChunks flag and re-test, must be empty
+        response = self._getJson(self.app.delete(tablesURI + '/' + tblName + "?dropChunks=yes"))
+        response = self._getJson(self.app.get(tablesURI))
+        tables = [obj['name'] for obj in response['results']]
+        self.assertEqual(tables, [])
+
+        # drop database
+        response = self._getJson(self.app.delete('/dbs/' + dbName))
+
+#
+#  run unit tests when imported as a main module
+#
+if __name__ == "__main__":
```

Comment:
Good job, I think this service will be very useful for Qserv data-loading, and eventually monitoring. You provide lots of major feature for data loading and it seems it can easily be extended. Thanks :-)
It lack a reasonable authentication mechanism, and also a autorisation mechanism (I don't want to multiply account but we might use an account for data loading and a restricted account for monitoring)